controller/core/src/instructions/internal_utils.rs

```diff
@@ -16,3 +16,21 @@ pub(crate) const fn caba<const A: usize, const START: usize, const LEN: usize>(
     }
     arr
 }
+
+#[inline]
+pub(crate) const fn split<T, const M: usize, const N: usize, const X: usize>(
+    data: &[T; M],
+) -> (&[T; N], &[T; X]) {
+    const {
+        assert!(N <= M);
+        assert!(X == M - N)
+    }
+
+    let (a, b) = data.split_at(N);
+
+    // SAFETY: data is guaranteed to be of length M
+    // and we are splitting it into two slices of length N and X (i.e M-N)
+    (unsafe { &*a.as_ptr().cast::<[T; N]>() }, unsafe {
+        &*b.as_ptr().cast::<[T; X]>()
+    })
+}
```

Comment:
nit: hmm i just realized, since we're likely only gonna use this for `T = u8` (exactly the inverse of `caba`), lets just remove the `T` generic and then we can rename this like `csba` for "const_split_byte_array". Saves us the trouble of writing `<u8,` everywhere